site/app/controllers/admin/ReportController.php

```diff
@@ -652,95 +684,209 @@ public function generateCustomization() {
+        // Do not add return type, for example if you add :MultiResponse
+        // php-stan will start yelling that missing return statement
+        // if we add return statement, it will be written into the downloaded file as json
+        // then it will break the Make remove_json_comments, so no linting is OK here.
+    }
+
+    #[Route("/courses/{_semester}/{_course}/reports/rainbow_grades_customization/gui_download", methods: ["GET"])]
+    public function downloadGUIRainbowConfig() {
         $rainbow_grades_dir = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "rainbow_grades");
+        $file_path = FileUtils::joinPaths($rainbow_grades_dir, 'gui_customization.json');
 
-        if (!move_uploaded_file($upload['tmp_name'], FileUtils::joinPaths($rainbow_grades_dir, 'customization.json'))) {
-            $msg = 'Upload failed: Could not copy file';
-            $this->core->addErrorMessage($msg);
+        if (!file_exists($file_path)) {
+            $msg = 'Download failed: File not found';
+            $redirect_url = $this->core->buildCourseUrl(['reports']);
             return new MultiResponse(
                 JsonResponse::getErrorResponse($msg),
                 null,
                 new RedirectResponse($redirect_url)
             );
         }
 
-        $msg = 'Rainbow Grades Customization uploaded';
-        $this->core->addSuccessMessage($msg);
-        return new MultiResponse(
-            JsonResponse::getSuccessResponse([
-                'customization_path' => $rainbow_grades_dir
-            ]),
-            null,
-            new RedirectResponse($redirect_url)
-        );
+        // Set headers and read the file
+        header('Content-Type: application/json');
+        header("Content-Transfer-Encoding: Binary");
+        header("Content-disposition: attachment; filename=\"" . basename($file_path) . "\"");
+
+        readfile($file_path);
+        exit;
+        // Do not add return type, for example if you add :MultiResponse
+        // php-stan will start yelling that missing return statement
+        // if we add return statement, it will be written into the downloaded file as json
+        // then it will break the Make remove_json_comments, so no linting is OK here.
     }
 
-    #[Route("/courses/{_semester}/{_course}/reports/rainbow_grades_status")]
+
+    #[Route('/courses/{_semester}/{_course}/reports/rainbow_grades_customization/manual_or_gui', methods: ['POST'])]
+    public function setRainbowGradeCustomization(): JsonResponse {
+
+        // Extract the value from $_POST
+        $selectedValue = $_POST['selected_value'] ?? null;
+
+        if (!isset($selectedValue) || trim($selectedValue) === '') {
+            $msg = 'Invalid request: No selected value provided.';
+            return JsonResponse::getErrorResponse($msg);
+        }
+
+        $rainbow_grades_dir = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "rainbow_grades");
+        $customization_src = null;
```

Comment:
This line isn't needed.  PHP uses function scoping, so the variable does not need to be defined outside the switch.